skrifa/src/outline/base.rs

```diff
@@ -0,0 +1,58 @@
+//! Common functionality for glyf, cff and autohinting scalers.
+
+use raw::{
+    tables::{hmtx::Hmtx, hvar::Hvar},
+    types::{BigEndian, F2Dot14, GlyphId, Tag},
+    FontRef, TableProvider,
+};
+
+/// Common functionality for glyf, cff and autohinting scalers.
+#[derive(Clone)]
+pub(crate) struct BaseOutlines<'a> {
+    pub font: FontRef<'a>,
+    pub hmtx: Hmtx<'a>,
+    pub hvar: Option<Hvar<'a>>,
+}
+
+impl<'a> BaseOutlines<'a> {
+    pub fn new(font: &FontRef<'a>) -> Option<Self> {
+        let hmtx = font.hmtx().ok()?;
+        let hvar = font.hvar().ok();
```

Comment:
perhaps worth a comment that we need hmtx, it's ok if there is no hvar

when the types don't popup it's very easy to miss that only one has a ?

Reply:
Done